### PR TITLE
Wl bl sub domain

### PR DESCRIPTION
**Pull request by stephdl**
Sort subdomains and domain, then expand them in separate map, the multimap has been modified to whitelist or blacklist subdomain 

https://github.com/NethServer/dev/issues/5940

### CODE REVIEW

File: filter/etc/e-smith/templates/etc/rspamd/local.d/multimap.conf/10base [hunk 1]
```diff
@@ -55,12 +70,27 @@ FROM_BLACKLIST {
...
+
 #blacklist the domains of senders
 FROM_DOMAINS_BLACKLIST {
     map = [
     "${CONFDIR}/blacklist_from_domains.map",
     ];
     type = "from";
+    require_symbols = "!FROM_SUBDOMAINS_BLACKLIST";
     filter = "email:domain:tld";
     description = "Refused list of FROM domains";
```
Comment (DavidePrincipi) [inline filter/etc/e-smith/templates/etc/rspamd/local.d/multimap.conf/10base]:
```suggestion
    description = "Reject SMTP sender by top level domain name suffix";
```

File: filter/etc/e-smith/templates/etc/rspamd/local.d/multimap.conf/10base [hunk 2]
```diff
@@ -55,12 +70,27 @@ FROM_BLACKLIST {
...
 }
 
+#blacklist the subdomains of senders
+FROM_SUBDOMAINS_BLACKLIST {
+    map = [
+    "${CONFDIR}/blacklist_from_subdomains.map",
+    ];
+    type = "from";
+    filter = "email:domain";
+    description = "Refused list of FROM subdomains";
```
Comment (DavidePrincipi) [inline filter/etc/e-smith/templates/etc/rspamd/local.d/multimap.conf/10base]:
```suggestion
    description = "Reject SMTP sender by exact domain name";
```

File: filter/etc/e-smith/templates/etc/rspamd/local.d/multimap.conf/10base [hunk 3]
```diff
@@ -27,13 +27,28 @@ FROM_WHITELIST {
...
+FROM_SUBDOMAINS_WHITELIST {
+
+    map = [
+    "${CONFDIR}/whitelist_from_subdomains.map",
+    ];
+    type = "from";
+    prefilter = true;
+    action = "accept";
+    filter = "email:domain";
+    description = "Accepted list of FROM subdomains";
```
Comment (DavidePrincipi) [inline filter/etc/e-smith/templates/etc/rspamd/local.d/multimap.conf/10base]:
```suggestion
    description = "Accept SMTP sender by exact domain name";
```

File: filter/etc/e-smith/templates/etc/rspamd/whitelist_to_subdomains.map/10base [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    if ($domain =~ m/@/){
+       next;
+    }
+    my $count = () = $domain =~ m/\./g;
+    push @whitelist, $domain if ($count > 1 );
+    }
+
+    foreach my $lines (@whitelist){
+    $OUT .= "$lines\n";
```
Comment (DavidePrincipi) [inline filter/etc/e-smith/templates/etc/rspamd/whitelist_to_subdomains.map/10base]: Fix indentation

```suggestion
        $OUT .= "$lines\n";
```

File: filter/etc/e-smith/templates/etc/rspamd/whitelist_to_domains.map/10base [hunk 5]
```diff
@@ -6,7 +6,12 @@
 {
 if(defined $rspamd{RecipientWhiteList}) {
     foreach my $domain (split(',', ($rspamd{RecipientWhiteList}))) {
-    push @whitelist, $domain unless ($domain =~ m/@/);
+
+    if ($domain =~ m/@/){
+       next;
+    }
+    my $count = () = $domain =~ m/\./g;
+    push @whitelist, $domain if ($count == 1 );
```
Comment (DavidePrincipi) [inline filter/etc/e-smith/templates/etc/rspamd/whitelist_to_domains.map/10base]:
```suggestion
        push @whitelist, $domain if ($count == 1 );
```

File: filter/etc/e-smith/templates/etc/rspamd/whitelist_from_subdomains.map/10base [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    
+        if ($domain =~ m/@/){
+           next;
+        }
+        my $count = () = $domain =~ m/\./g;
+        push @whitelist, $domain if ($count > 1 );
+    }
+
+    foreach my $lines (@whitelist){
+    $OUT .= "$lines\n";
```
Comment (DavidePrincipi) [inline filter/etc/e-smith/templates/etc/rspamd/whitelist_from_subdomains.map/10base]:
```suggestion
        $OUT .= "$lines\n";
```

File: filter/etc/e-smith/templates/etc/rspamd/whitelist_from_domains.map/10base [hunk 7]
```diff
@@ -6,7 +6,12 @@
 {
 if(defined $rspamd{SenderWhiteList}) {
     foreach my $domain (split(',', ($rspamd{SenderWhiteList}))) {
-    push @whitelist, $domain unless ($domain =~ m/@/);
+    
+        if ($domain =~ m/@/){
+           next;
+        }
+        my $count = () = $domain =~ m/\./g;
+        push @whitelist, $domain if ($count == 1 );
```
Comment (DavidePrincipi) [inline filter/etc/e-smith/templates/etc/rspamd/whitelist_from_domains.map/10base]:
```suggestion
            push @whitelist, $domain if ($count == 1 );
```

**Comment by stephdl** [comment]
> What do you think?

agreed, just some minor differences on indentation, can I merge and go to testing ?

**Pull request merged**